extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/AuthenticationProviderService.java

```diff
@@ -97,13 +106,17 @@ private CredentialsInfo getRadiusChallenge(RadiusPacket challengePacket) {
         }
 
         // We have the required attributes - convert to strings and then generate the additional login box/field
-        String replyMsg = replyAttr.toString();
+        logger.debug("Received challenge: {}", replyAttr.toString());
+        String replyMsg = replyAttr.toString().split(" = ", 2)[1];
```

Comment:
Would `replyAttr.getValue()` work here? Formatting the `RadiusAttribute` as a string and then re-parsing it back into an attribute feels like it shouldn't be necessary.

Reply:
Well, two things, here:
* In order to get the "Reply-Message = " out of here, I need to split it.  Not sure how feasible this is with an `AttributeValue` type?
* It ends up getting passed through the form on the Web side.  Is this going to play okay with using `AttributeValue` vs. `String`?

Reply:
Checking the source of `RadiusAttribute`, [the `toString()` function](https://github.com/coova/jradius/blob/master/core/src/main/java/net/jradius/packet/attribute/RadiusAttribute.java#L155-L158) says:

```java
public String toString()
{
    return attributeName + " " + Operator.operatorToString(attributeOp) + " " + attributeValue.toString();
}
```

with `attributeValue` being exposed through [`getValue()`](https://github.com/coova/jradius/blob/4ee91e33fa155843fe428559d81f036051d30515/core/src/main/java/net/jradius/packet/attribute/RadiusAttribute.java#L94-L97):

```java
public AttributeValue getValue()
{
    return attributeValue;
}
```

From the above, it looks like calling `replyAttr.toString()` and splitting that to obtain the value is equivalent to `replyAttr.getValue().toString()` (the value used in the `logger.debug()` call), with the latter avoiding the back-and-forth conversion and assumption of the contents of `attributeOp`.

Is this not the case?

Reply:
Ah, okay, i see.  I'll give it a shot.